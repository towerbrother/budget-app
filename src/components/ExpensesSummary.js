import React from "react";
import { connect } from "react-redux";
import selectExpenses from "../selectors/expenses";
import getExpensesTotal from "../selectors/expenses-total";
import numeral from "numeral";

export const ExpensesSummary = (props) => {
  const total = numeral(props.expensesTotal / 100).format("$0,0.00");
  return (
    <div>
      <p>
        {props.expensesCount === 0
          ? `No expenses`
          : props.expensesCount === 1
          ? `Viewing 1 expense totalling ${total}`
          : `Viewing ${props.expensesCount} expenses totalling ${total}`}
      </p>
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    expensesCount: selectExpenses(state.expenses, state.filters).length,
    expensesTotal: getExpensesTotal(
      selectExpenses(state.expenses, state.filters)
    ),
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
