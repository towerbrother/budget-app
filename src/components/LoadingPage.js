import React from "react";

const LoadingPage = () => {
  return (
    <div className="loader">
      <img alt="loader gif" className="loader__image" src="/loader.gif" />
    </div>
  );
};

export default LoadingPage;
