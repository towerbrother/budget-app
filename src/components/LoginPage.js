import React from "react";

const LoginPage = () => {
  return (
    <div>
      <button>Login</button>
    </div>
  );
};

export default LoginPage;
