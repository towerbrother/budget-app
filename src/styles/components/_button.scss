.button {
  background: $blue;
  border: none;
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  margin-bottom: $s-size;
  padding: $s-size;
  text-decoration: none;
}

.button__link {
  background: none;
}

.button__secondary {
  background: #c2300c;
}
