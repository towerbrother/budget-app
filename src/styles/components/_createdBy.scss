.createdBy__text {
  color: black;
  display: block;
  font-size: $m-size;
  margin: $m-size;
  text-align: center;
}

.createdBy__link {
  color: black;
  text-decoration: underline;
}
