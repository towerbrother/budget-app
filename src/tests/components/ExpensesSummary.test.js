import React from "react";
import { ExpensesSummary } from "./../../components/ExpensesSummary";
import expensesTestData from "../fixtures/expenses";
import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

test("should render ExpensesSummary correctly with multiple expenses", () => {
  const wrapper = shallow(<ExpensesSummary expenses={expensesTestData} />);
  expect(wrapper).toMatchSnapshot();
});

test("should render ExpensesSummary correctly with one expense", () => {
  const wrapper = shallow(<ExpensesSummary expenses={expensesTestData[0]} />);
  expect(wrapper).toMatchSnapshot();
});
