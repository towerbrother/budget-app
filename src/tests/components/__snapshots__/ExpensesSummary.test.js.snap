// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly with multiple expenses 1`] = `ShallowWrapper {}`;

exports[`should render ExpensesSummary correctly with one expense 1`] = `ShallowWrapper {}`;
